src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/PublishingServiceTests.cs

```diff
@@ -31,23 +31,85 @@ public void RetryStage()
                 PreviousVersionId = new Guid("af032e3c-67c2-4562-9717-9a305a468263")
             };
 
-            using (var context = InMemoryApplicationDbContext("RetryContentAndPublishing"))
+            using (var context = InMemoryApplicationDbContext("RetryReleaseStage"))
```

Comment:
There is a reason why we do this. When you say "we don't (usually) need to create multiple context scopes" I'm not sure if you are acknowledging this but I'll explain anyway just in case.

Imagine the service under test has this method:

```
        private async Task<string> GetTopicTitle(ContentDbContext context, Guid publicationId)
        {
            // Bug here - code doesn't add Topic navigation property to Queryable
            var publication = await context.Publications.FindAsync(publicationId);
            return publication.Topic.Title;
        }
```

This test highlights the bug and fails:

```
        [Fact]
        public async Task GetTopicTitle_ReturnsTitle()
        {
            var publication = new Publication
            {
                Id = Guid.NewGuid(),
                Topic = new Topic
                {
                    Id = new Guid(),
                    Title = "Ben test"
                }
            };

            using (var context = InMemoryApplicationDbContext("GetTopicTitle"))
            {
                context.Add(publication);
                await context.SaveChangesAsync();
            }

            using (var context = InMemoryApplicationDbContext("GetTopicTitle"))
            {
                // Service call
                // Fails with System.NullReferenceException : Object reference not set to an instance of an object.
                var result = await GetTopicTitle(context, publication.Id);
                
                Assert.Equal("Ben test", result);
            }
        }
```

It fails before the assertion with`System.NullReferenceException : Object reference not set to an instance of an object.` as expected.

While this test as you describe will succeed:

```
        [Fact]
        public async Task GetTopicTitle_ReturnsTitle()
        {
            var publication = new Publication
            {
                Id = Guid.NewGuid(),
                Topic = new Topic
                {
                    Id = new Guid(),
                    Title = "Ben test"
                }
            };

            var context = InMemoryApplicationDbContext();

            context.Add(publication);
            await context.SaveChangesAsync();
            
            var result = await GetTopicTitle(context, publication.Id);
            
            Assert.Equal("Ben test", result);
            // Success!
        }
```

Reply:
Ah I see, fair enough! Pretty crappy that the context will 'helpfully' load in the entity graph when you don't actually want it, but won't do it by default when you need it!

I'll have to re-work some of my tests to keep this in mind.

Reply:
Yes it's been the cause of a few bugs in the application! 🐛